File: frontend/src/types/usuario.ts
export interface Rol {
  id: number;
  nombre: string;
  descripcion: string;
}

export interface Usuario {
  id: number;
  nombre: string;
  correo: string;
  roles: Rol[];
  creado: string;
  habilitado: boolean;
  fotoPerfil?: string;
}

export interface RegistroRequest {
  nombre: string;
  correo: string;
  clave: string;
  rolesIds: number[];
}

export interface LoginRequest {
  correo: string;
  clave: string;
}

export interface LoginResponse {
  token: string;
  usuario: Usuario;
}

export interface Perfil {
  id: number;
  nombre: string;
  correo: string;
  fotoPerfil: string;
  roles: string[];
}

export interface EditarPerfilRequest {
  nombre: string;
}

File: backend/src/main/java/com/TuraTrip/backend/models/Usuario.java
package com.TuraTrip.backend.models;

import java.time.LocalDateTime;
import java.util.HashSet;
import java.util.Set;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.JoinTable;
import jakarta.persistence.ManyToMany;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "usuarios")
@Getter 
@Setter 
@NoArgsConstructor 
@AllArgsConstructor 
@Builder
public class Usuario {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 100)
    private String nombre;

    @Column(nullable = false, unique = true, length = 255)
    private String correo;

    @Column(nullable = false, length = 255)
    private String clave;

    @Column(name = "foto_perfil", length = 500)
    private String fotoPerfil;

    @Builder.Default
    @Column(nullable = false)
    private Boolean habilitado = false;

    @Builder.Default
    @ManyToMany(fetch = FetchType.EAGER)
    @JoinTable(
        name = "usuario_roles",
        joinColumns = @JoinColumn(name = "usuario_id"),
        inverseJoinColumns = @JoinColumn(name = "rol_id")
    )
    private Set<Rol> roles = new HashSet<>();

    // Nueva Relación: Un usuario tiene muchas publicaciones
    @Builder.Default
    @OneToMany(mappedBy = "usuario", cascade = CascadeType.ALL, orphanRemoval = true)
    private Set<Publicacion> publicaciones = new HashSet<>();

    @CreationTimestamp
    @Column(updatable = false)
    private LocalDateTime creado;

    @UpdateTimestamp
    private LocalDateTime actualizado;
}